### ISSUE TITLE
Async Iterators

### ISSUE THREAD

**Issue by RangerMauve** [leebyron/ecmascript-iterator-hof#2]
Would it make sense for this proposal to also define the same functions for Async Iterables to kill two birds with one stone?

Having higher order functions on async iterables would make async programming extremely powerful.

**Comment by leebyron** [comment]
Yeah - I think these functions definitely deserve to belong on both typical and async iterators (should those reach stage 4) - not sure if that should be two separate proposals or one collective proposal.

**Comment by RangerMauve** [comment]
Since Async Iterators are right on the [cusp of being implemented in v8](https://github.com/tc39/proposal-async-iteration/issues/63), I think it's safe to say that they will be part of the standard before this proposal hits. If it's possible, I think it would make sense to include Async iterators at the same time to avoid confusion.

**Comment by zimme** [comment]
The only reason I see for these to be separate proposals is if any of the hofs would be different for the iterator types.

**Comment by RangerMauve** [comment]
Now that async-iterators are available pretty much everywhere, there shouldn't be much question about it.
